kalite/main/migrations/0039_log_timestamps.py

```diff
@@ -0,0 +1,246 @@
+# -*- coding: utf-8 -*-
+from south.utils import datetime_utils as datetime
+from south.db import db
+from south.v2 import DataMigration
+from django.db import models
+
+class Migration(DataMigration):
+
+    def forwards(self, orm):
+        """
+        Create latest_activity_timestamp data when possible by inspecting the model.
+        For ExerciseLog inspect relevant AttemptLogs to infer most recent activity.
+        If not available, and for other Logs, fallback to completion_timestamp if available.
+        """
+        for e in orm.ExerciseLog.objects.filter(latest_activity_timestamp=None):
```

Comment:
Would be good to add a surrounding `try...except` around everything here, from my experience with weird missing users in running similar code on the ExerciseLogs on globe (with `e.user` not pointing to a valid user).